template/README.md

```diff
@@ -0,0 +1,11 @@
+# DAO template
+
+This package serves as a template for writing new DAO contracts.
+
+## How to write your own DAOs
+
+To create your own DAO contract:
+1. Either fork this repository, or copy the [`template`](/template) folder to your repository;
```

Comment:
Now I see that copying the `template` folder is not so straightforward. It's not enough to just copy this folder, one also needs `stack.yaml` file to make it work, in particular they need to point to the `baseDAO` package somehow. So probably we should provide a custom `stack.yaml` file here. Note that we can create `resolver.yaml` (or `snapshot.yaml`) file with common parts of the root `stack.yaml` and this new custom `stack.yaml`.

This custom `stack.yaml` needs to point to this repo, ideally it should point to the current commit, but it's impossible. Maybe we can have `stack.yaml` here that looks like:
```
resolver: resolver.yaml
extra-deps:
- git:
    https://github.com/bla-bla # baseDAO link
  commit:
    # TODO
```
and ask the user to provide last commit instead of `TODO`.

If this single extra-dep is the only addition to the root `stack.yaml`, probably we can only use the root one and treat this custom one as an additional artifact.

Reply:
> Note that we can create resolver.yaml (or snapshot.yaml) file with common parts of the root stack.yaml and this new custom stack.yaml.

If I do so, where should I put it? Looks like I shouldn't put the snapshot into the root of `baseDAO`, because then again the user won't be able to just copy-paste the folder and will get a confusing error about `../resolver.yaml` not existing.

> If this single extra-dep is the only addition to the root stack.yaml, probably we can only use the root one and treat this custom one as an additional artifact.

Do you mean I should try to put an extra dep refering to `baseDAO` in `stack.yaml` of `baseDAO`?

Reply:
No, I think we can have `resolver.yaml` and `stack.yaml` in `template/`. `template/resolver.yaml` can be used in both `stack.yaml` files. The root one doesn't need anything else. The one in `template/` also needs `baseDAO` in its `extra-deps`.

AFAIU `extra-deps` listed in `stack.yaml` will be concatenated with `extra-deps` from `resolver.yaml`.

Reply:
Okay, I implemented this suggestion.

The only problem with it which I see is that if we leave commit sha uninitialized in `template/stack.yaml`'s `extra-deps`, then building the template project will be tiresome for developers who introduce breaking changes, and for CI.

Reply:
Aha, my idea was to not use `template/stack.yaml` to build `template` in **this** repo. I. e. keep `template` in `packages` in root `stack.yaml` and use the root `stack.yaml` for building. This way when we build **this** repo we will take `baseDAO` from **this** repo. I think it should work fine as long as `baseDAO` extra-dep is the only difference between two of them. I mean, if it builds for us with the root `stack.yaml`, it should also build for end user with `template/stack.yaml` as long as they provide the right revision.

But apparently it's complicated because `stack` uses the deepest `stack.yaml` it sees. One option is to make user's life a bit harder by renaming `template/stack.yaml` in this repo to `template/_stack.yaml` and asking the user to rename it. Or by asking them to pass `--stack-yaml` or `--resolver`. But probably user's convenience is more important and we should handle this problem on our side. It's possible to point `stack` to the desired `stack.yaml`, we just need to do it in all places where we `stack build`:
1. Make CI build both packages using the root `stack.yaml`. I don't know precisely how to do because we are using `haskell.nix` there, but probably you'll figure it out or will ask our nix experts.
2. Developers are supposed to use `Makefile`. There we currently build only `baseDAO` and it should work fine. We should add a way to build `template` and this command should point to the right `stack.yaml`.

Does it sound alright?

Reply:
> Aha, my idea was to not use template/stack.yaml to build template in this repo. I. e. keep template in packages in root stack.yaml and use the root stack.yaml for building.

Oh okay. I first tried this and got some weird error, but probably this error was caused by a different reason. If I add `template` to `/stack.yaml`, then everything works for me with just `stack build`.

> But apparently it's complicated because stack uses the deepest stack.yaml it sees.

Does it work for you so, or you are referring to my experience? Looks like for me `stack` always thinks that `/stack.yaml` is the one to work with, unless I'm within `template` folder.

Reply:
> Does it work for you so, or you are referring to my experience? 

I thought that it works this way, but I didn't test it. Also I thought that it worked this way for you.

> Looks like for me stack always thinks that /stack.yaml is the one to work with, unless I'm within template folder.

Okay, cool then, can you add `template` back to the root `stack.yaml`, so that I could test that version? Also it'd be nice to rebase on master (there is a conflict) and update `Makefile`.

Reply:
Yep, should be fixed now.

Reply:
Okay, `make` seems to work fine now. However, `stack build` doesn't work from the `template` folder, which is expected and fine, but I think it's worth mentioning somewhere, maybe in `CONTRIBUTING.md`.